lib/streamlit/legacy_caching/hashing.py

```diff
@@ -712,14 +712,14 @@ def _code_to_bytes(self, code, context: Context, func=None) -> bytes:
 
     @staticmethod
     def _get_main_script_directory() -> str:
-        """Get the directory of the main script."""
+        """Get the absolute path to directory of the main script."""
         import __main__
-        import os
+        import pathlib
 
         # This works because we set __main__.__file__ to the
         # script path in ScriptRunner.
-        main_path = __main__.__file__
-        return str(os.path.dirname(main_path))
+        abs_main_path = pathlib.Path(__main__.__file__).absolute()
```

Comment:
I think we should change this to `.resolve()` given that `.absolute()` is undocumented.

Reply:
Hey @amhrasmussen, heads up that I had some free time over the past few days so decided to spend some of it helping push a bunch of open PRs through.

This requested change was simple enough that I figured I might as well wrap it up given you're busy enjoying your summer 🙂